diff -ur mp3gain-1_4_6-src.zip_FILES.OLD/mp3gain.c mp3gain-1_4_6-src.zip_FILES/mp3gain.c
--- mp3gain-1_4_6-src.zip_FILES.OLD/mp3gain.c	2004-12-23 15:47:04.000000000 -0500
+++ mp3gain-1_4_6-src.zip_FILES/mp3gain.c	2006-11-20 17:14:03.000000000 -0500
@@ -42,7 +42,11 @@
  *  out there will have come up with superior versions already...
  *
  *  So have fun dissecting.
- */
+ */
+
+/* stf: make 	-t the default
+ *		-T restores old behaviour
+ */
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -117,8 +121,8 @@
 unsigned char buffer[BUFFERSIZE];
 
 int writeself = 0;
-int QuietMode = 0;
-int UsingTemp = 0;
+int QuietMode = 0;
+int UsingTemp = 1;
 int NowWriting = 0;
 double lastfreq = -1.0;
 
@@ -701,7 +705,8 @@
 		        fclose(inf); 
 			inf = NULL;
             passError(MP3GAIN_UNSPECIFED_ERROR, 3,
-                "\nCan't open ", outfilename, " for temp writing\n");
+                "\nCan't open ", outfilename, " for temp writing\n");
+			NowWriting = 0;
 			return M3G_ERR_CANT_MAKE_TMP;
 		} 
  
@@ -715,7 +720,8 @@
 	  if (UsingTemp && (outf != NULL))
 		  fclose(outf);
 	  passError( MP3GAIN_UNSPECIFED_ERROR, 3,
-          "\nCan't open ", filename, " for modifying\n");
+          "\nCan't open ", filename, " for modifying\n");
+	  NowWriting = 0;
 	  return M3G_ERR_CANT_MODIFY_FILE;
   }
   else {
@@ -926,7 +932,8 @@
 			passError(MP3GAIN_CANCELLED,3,"Cancelled processing.\n", filename, " is probably corrupted now.");
 		}
 		if (saveTime) 
-		  fileTime(filename, setStoredTime);		
+		  fileTime(filename, setStoredTime);
+		NowWriting = 0;		
 		return;
 	}
 #endif
@@ -995,8 +1002,9 @@
         if (outlength != inlength) {
             deleteFile(outfilename);
 			passError( MP3GAIN_UNSPECIFED_ERROR, 3,
-                "Not enough temp space on disk to modify ", filename, 
-                "\nEither free some space, or do not use \"temp file\" option\n");
+                "Not enough temp space on disk to modify ", filename,
+		"\nEither free some space, or switch off \"temp file\" option with -T\n");
+            NowWriting = 0; 
             return M3G_ERR_NOT_ENOUGH_TMP_SPACE;
         }
         else {
@@ -1004,7 +1012,8 @@
 		    if (deleteFile(filename)) {
 				deleteFile(outfilename); //try to delete tmp file
 				passError( MP3GAIN_UNSPECIFED_ERROR, 3,
-                    "Can't open ", filename, " for modifying\n");
+                    "Can't open ", filename, " for modifying\n");
+			    NowWriting = 0;
 			    return M3G_ERR_CANT_MODIFY_FILE;
 		    }
 		    if (moveFile(outfilename, filename)) {
@@ -1012,7 +1021,8 @@
                     "Problem re-naming ", outfilename, " to ", filename, 
                     "\nThe mp3 was correctly modified, but you will need to re-name ", 
                     outfilename, " to ", filename, 
-                    " yourself.\n");
+                    " yourself.\n");
+			    NowWriting = 0;
 			    return M3G_ERR_RENAME_TMP;
 		    };
 		    if (saveTime)
@@ -1214,9 +1224,10 @@
 		fprintf(stderr,"\t%cm <i> - modify suggested MP3 gain by integer i\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cd <n> - modify suggested dB gain by floating-point n\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cc - ignore clipping warning when applying gain\n",SWITCH_CHAR);
-		fprintf(stderr,"\t%co - output is a database-friendly tab-delimited list\n",SWITCH_CHAR);
-		fprintf(stderr,"\t%ct - mp3gain writes modified mp3 to temp file, then deletes original\n",SWITCH_CHAR);
-		fprintf(stderr,"\t     instead of modifying bytes in original file\n");
+		fprintf(stderr,"\t%co - output is a database-friendly tab-delimited list\n",SWITCH_CHAR);
+		fprintf(stderr,"\t%ct - mp3gain writes modified mp3 to temp file, then deletes original \n",SWITCH_CHAR);
++		fprintf(stderr,"\t     instead of modifying bytes in original file (default)\n");
++		fprintf(stderr,"\t%cT - mp3gain directly modifies mp3 file (opposite of %ct)\n",SWITCH_CHAR,SWITCH_CHAR);
 		fprintf(stderr,"\t%cq - Quiet mode: no status messages\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cp - Preserve original file timestamp\n",SWITCH_CHAR);
 		fprintf(stderr,"\t%cx - Only find max. amplitude of mp3\n",SWITCH_CHAR);
@@ -1505,9 +1516,12 @@
 
                     break;
 
-				case 't':
-				case 'T':
+				case 't':
 					UsingTemp = !0;
+					break;
+
+				case 'T':
+					UsingTemp = 0;
 					break;
 
 				case 'u':
